frontend/src/app/PrimeErrorBoundary.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ErrorInfo } from "react";
+
+import ErrorPage from "./commonComponents/ErrorPage";
+import { getAppInsights } from "./TelemetryService";
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+}
+
+export default class PrimeErrorBoundary extends React.Component<
+  {},
+  ErrorBoundaryState
+> {
+  constructor(props: any) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true };
+  }
+
+  componentDidCatch(error: Error, info: ErrorInfo) {
+    const appInsights = getAppInsights();
+    if (appInsights) {
+      appInsights.trackException({
+        exception: new Error("PrimeErrorBoundary: " + error),
+        severityLevel: 3,
+        properties: { componentStack: info.componentStack },
```

Comment:
I know we used to have an alert on this, though I don't know if it's still active. Do you know if wrapping the component around the `Switch` instead of the `App` will cause any issues with tracing the root cause of an issue?

https://github.com/CDCgov/prime-simplereport/pull/2014

Reply:
AFAIK, I think this will just add one or two more components to the stack (`Switch`, 'Route') - but the trace should still lead back to where the error was thrown!